service_grpc_namespace/src/lib.rs

```diff
@@ -313,6 +316,71 @@ mod tests {
         });
     }
 
+    #[tokio::test]
+    async fn test_invalid_namespace_name_error() {
+        let catalog: Arc<dyn Catalog> =
+            Arc::new(MemCatalog::new(Arc::new(metric::Registry::default())));
+
+        let topic = catalog
+            .repositories()
+            .await
+            .topics()
+            .create_or_get("kafka-topic")
+            .await
+            .unwrap();
+        let query_pool = catalog
+            .repositories()
+            .await
+            .query_pools()
+            .create_or_get("query-pool")
+            .await
+            .unwrap();
+
+        let handler = NamespaceService::new(catalog, Some(topic.id), Some(query_pool.id));
+
+        // Length constraint
+        let name = "A".repeat(1000);
+        let invalid_name_err = NamespaceNameError::LengthConstraint {
+            name: String::from(&name),
+        };
+        let req = CreateNamespaceRequest {
+            name,
+            retention_period_ns: Some(RETENTION),
+        };
+        let create_ns_response = handler.create_namespace(Request::new(req)).await;
+        assert_matches!(
+            create_ns_response,
+        s => match s {
+            Err(s) => {
+                assert_eq!(s.code(), Code::InvalidArgument);
+                assert_eq!(s.message(), invalid_name_err.to_string());
+            },
+            Ok(_) => panic!("NamespaceName was successful, when it should have failed."),
+        });
```

Comment:
I think you could probably simplify the match here if you fancied, though you would lose that panic message.

```suggestion
        assert_matches!(
            create_ns_response,
        Err(s) => {
            assert_eq!(s.code(), Code::InvalidArgument);
            assert_eq!(s.message(), invalid_name_err.to_string());
        });
```

Reply:
Yeah, I was on the fence about how verbose to be. Will make the change.